autorest/adal/token.go

```diff
@@ -247,6 +259,47 @@ func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID s
 	)
 }
 
+func NewServicePrincipalTokenFromMSI(oauthConfig OAuthConfig, resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
+	// Read MSI settings
+	bytes, err := ioutil.ReadFile("/var/lib/waagent/ManagedIdentity-Settings")
```

Comment:
I would provide this file as a *Reader* argument to this function. The name or the path may change in the future and this will provide more flexibility.